frontend/src/components/UpdateModal.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState, FunctionComponent, useEffect, useMemo } from "react";
+import { Button } from "react-bootstrap";
+import Modal from "react-bootstrap/Modal";
+import { AgGridReact } from "ag-grid-react";
+import "ag-grid-enterprise";
+import styles from "../pages/requests/requests.module.scss";
+import "ag-grid-community/styles/ag-grid.css";
+import "ag-grid-community/styles/ag-theme-alpine.css";
+import { CellChange } from "../pages/requests/helpers";
+
+export const UpdateModal: FunctionComponent<{
+  changes: CellChange[];
+  onHide: () => void;
+}> = ({ changes, onHide }) => {
+  const [rowData, setRowData] = useState(changes);
+  const [columnDefs] = useState([
+    { field: "primaryId", rowGroup: true, hide: true },
+    { field: "field" },
+    { field: "oldValue" },
+    { field: "newValue" },
+  ]);
+
+  useEffect(() => {
+    setRowData(changes);
+  }, [changes]);
+
+  const autoGroupColumnDef = useMemo(() => {
+    return {
+      headerName: "Primary Id",
+      field: "primaryId",
+    };
+  }, []);
+
+  return (
+    <Modal
+      show={true}
+      size={"lg"}
+      centered
+      onHide={onHide}
+      className={styles.overlay}
+    >
+      <Modal.Header closeButton>
+        <Modal.Title id="contained-modal-title-vcenter">
+          Are you sure?
+        </Modal.Title>
+      </Modal.Header>
+
+      <Modal.Body>
+        <p>Are you sure you want to submit the following changes?</p>
+        <div className="ag-theme-alpine" style={{ height: 350 }}>
+          <AgGridReact
+            rowData={rowData}
+            columnDefs={columnDefs}
+            groupRemoveSingleChildren={true}
+            autoGroupColumnDef={autoGroupColumnDef}
+            groupDefaultExpanded={1}
+          ></AgGridReact>
+        </div>
```

Comment:
on confirmation of changes, this hook should be called per sample / set of changes

https://github.com/mskcc/smile-dashboard/pull/65/files#diff-cf60dcd267836c17e6d88e62f7d84788e8bbd7bce725647d722e01df94ac638bR126

the hook is `useUpdateSamplesMutation`

example usage

```
 * const [updateSamplesMutation, { data, loading, error }] = useUpdateSamplesMutation({
 *   variables: {
 *      where: // value for 'where'
 *      update: // value for 'update'
 *      connect: // value for 'connect'
 *   },
 * });
 * 
 *```

Reply:
Example in use by test page `UpdateSamples.tsx`

```
  const [updateSamplesMutation, { ...args }] = useUpdateSamplesMutation({
    variables: {
      where: {
        smileSampleId: params.smileSampleId,
      },
      update: {
        revisable: false,
        hasMetadataSampleMetadata: [
          {
            update: {
              node: {
                cmoPatientId: "C-NEWPATIENTID",
                investigatorSampleId: "new investigator id",
                // "sampleClass": null,
                // "sampleName": null,
                // "sampleOrigin": null,
                // "sampleType": null,
                // "tumorOrNormal": null
              },
            },
          },
        ],
      },
    },
  });
```